lib/DDGC/Web/Service/Blog.pm

```diff
@@ -297,13 +330,15 @@ sub post_update_or_create {
         $post = rset('User::Blog')->find( $params->{id} );
         bailout( 404, sprintf "Blog post %s not found", $params->{id} )
             if (!$post);
-
-        bailout( 403, "You do not have permission to update this post" )
-            if ($post->users_id != $params->{users_id});
     }
 
     try {
+        my $this_is_a_new_post = 0;
+        if ( (!$params->{id} && $params->{live}) || ( $post && !$post->live && $params->{live} ) ) { # new post
+            $this_is_a_new_post = 1;
+        }
         $post = rset('User::Blog')->update_or_create( $params );
+        eval '_add_blog_post_notification_bits( $post ) if $this_is_a_new_post';
```

Comment:
Because we do not care if it fails. It's not likely to, but I don't understand all the corners of this thing and our error handling is not graceful. Combine this with our caching policy (i.e. don't) and we'll have an angry customer if this goes wrong - lost blog post, lost time.